packages/core/src/interpreter.ts

```diff
@@ -399,16 +395,35 @@ export class Interpreter<
             State.from(initialState, this.machine.context)
           );
 
-    this.scheduler.initialize(() => {
-      this.update(resolvedState, initEvent as SCXML.Event<TEvent>);
+    this._state = resolvedState;
+
+    this.update(resolvedState);
+
+    if (this.options.devTools) {
+      this.attachDevTools();
+    }
+
+    this.flush();
 
-      if (this.options.devTools) {
-        this.attachDevTools();
-      }
-    });
     return this;
   }
 
+  private flush() {
+    this.mailbox.status = 'processing';
+    let event = this.mailbox.dequeue();
```

Comment:
This implementation doesn't scale too well with multiple consumers of the mailbox because a lot of the delicate queueing/dequeuing management becomes the responsibility of each consumer. I think I've managed to avoid this in this PR:
https://github.com/statelyai/xstate/pull/2916/commits/37547c356cf5f6fe5e073c9c5e1df57e4ddcb43b

With these changes, the consumer only needs to provide a "process"  callback to the created mailbox and call simple `start`, `clear` and `enqueue` methods. Thanks to that I've been able to reuse easily this Mailbox implementation in other 2 places that were managing this:
https://github.com/statelyai/xstate/pull/2916/commits/00f1d74d13d8105c7ebb52e771a82098b327ba79
https://github.com/statelyai/xstate/pull/2916/commits/e530b5b94c147e68fe3d51eea38d83469d8d49ac